zone/zoning.cpp

```diff
@@ -982,26 +1039,28 @@ void Client::SendZoneFlagInfo(Client *to) const {
 	to->Message(Chat::White, "Flags for %s:", GetName());
 
 	for(; cur != end; ++cur) {
-		uint32 zoneid = *cur;
-
-		const char *short_name = ZoneName(zoneid);
-
-		char *long_name = nullptr;
-		content_db.GetZoneLongName(short_name, &long_name);
-		if(long_name == nullptr)
-			long_name = empty;
-
-		float safe_x, safe_y, safe_z;
-		int16 minstatus = 0;
-		uint8 minlevel = 0;
+		uint32 zone_id = *cur;
+		const char* zone_short_name = ZoneName(zone_id);
+		std::string zone_long_name = zone_store.GetZoneLongName(zone_id);
+		float safe_x, safe_y, safe_z, safe_heading;
+		int16 min_status = 0;
+		uint8 min_level = 0;
 		char flag_name[128];
-		if(!content_db.GetSafePoints(short_name, 0, &safe_x, &safe_y, &safe_z, &minstatus, &minlevel, flag_name)) {
+		if(!content_db.GetSafePoints(
+			zone_short_name,
+			0,
+			&safe_x,
+			&safe_y,
+			&safe_z,
+			&safe_heading,
+			&min_status,
+			&min_level,
+			flag_name
+		)) {
 			strcpy(flag_name, "(ERROR GETTING NAME)");
 		}
 
-		to->Message(Chat::White, "Has Flag %s for zone %s (%d,%s)", flag_name, long_name, zoneid, short_name);
-		if(long_name != empty)
-			delete[] long_name;
+		to->Message(Chat::White, "Has Flag %s for zone %s (%d,%s)", flag_name, zone_long_name, zone_id, zone_short_name);
```

Comment:
Needs to be `zone_long_name.c_str()` in the log message

Reply:
Fixed.